.github/workflows/docs-preview-build.yml

```diff
@@ -0,0 +1,49 @@
+
+# This workflow builds the docs for each PR (including from forks), and uploads
+# the rendered docs as an artifact.
+#
+# The artifact will be picked up by docs-preview-deploy.yml and uploaded to
+# https://jj-preview.github.io/docs/.
+#
+# The split into "build" and "deploy" follows GitHub's guide at
+# https://securitylab.github.com/resources/github-actions-preventing-pwn-requests/.
+# The build workflow has no access to the repo secrets. The deploy workflow runs
+# in the context of the main repo rather than the PR branch.
+
+name: docs-preview-build
+
+on:
+  pull_request:
```

Comment:
Will this trigger when a pull request changes? If not, is it possible to provide a way to trigger it manually?